include/zephyr/drivers/flash.h

```diff
@@ -128,6 +128,10 @@ typedef int (*flash_api_read_jedec_id)(const struct device *dev, uint8_t *id);
 typedef int (*flash_api_ex_op)(const struct device *dev, uint16_t code,
 			       const uintptr_t in, void *out);
 
+#if defined(CONFIG_FLASH_RESET)
+typedef void (*flash_api_send_flash_reset)(const struct device *dev);
```

Comment:
No need to duplicate "flash" in the API name.
```suggestion
typedef void (*flash_api_send_reset)(const struct device *dev);
```